redfish-core/lib/processor.hpp

```diff
@@ -1436,6 +1490,60 @@ inline void requestRoutesOperatingConfig(App& app)
     });
 }
 
+inline void
+    handleProcessors(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                     const boost::system::error_code& ec,
+                     const std::vector<std::string>& objects)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    nlohmann::json& members = asyncResp->res.jsonValue["Members"];
+    members = nlohmann::json::array();
+
+    for (const auto& object : objects)
+    {
+        sdbusplus::message::object_path path(object);
+        std::string leaf;
+
+        /**
+         * @brief Workaround to handle DCM (Dual-Chip Module)
+         *        package for Redfish
+         *
+         * Make sure processor modeled as dual chip module,
+         * If yes then, replace redfish processor id as
+         * "dcmN-cpuN" because redfish does not support chip
+         * module concept.
+         *
+         * @note Inventory modeled as "dcmN/cpuN" so wherever
+         *       using redfish processor id as "dcmN-cpuN" then
+         *       that need to convert as "dcmN/cpuN" before
+         *       validating the inventory processor object path
+         */
+        if (path.parent_path().filename().find("dcm") != std::string::npos)
+        {
+            leaf = path.parent_path().filename() + "-" + path.filename();
+        }
+        else
+        {
+            leaf = path.filename();
+        }
+
+        if (leaf.empty())
+        {
+            continue;
+        }
+        std::string newPath = "/redfish/v1/Systems/system/Processors";
+        newPath += '/';
+        newPath += leaf;
```

Comment:
As it is URL, 
```
        std::string newPath = boost::urls::format("/redfish/v1/Systems/system/Processors/{}", leaf);
```

Reply:
Good point. I've also changed leaf above to use std::format() for clarity as well.